src/sc_kpm/sc_agent.py

```diff
@@ -14,40 +15,37 @@
 from sc_client.models import ScAddr, ScEvent, ScEventParams
 
 from sc_kpm.identifiers import QuestionStatus
-from sc_kpm.logging import get_kpm_logger
 from sc_kpm.sc_keynodes import Idtf, ScKeynodes
 from sc_kpm.sc_result import ScResult
 from sc_kpm.utils.action_utils import check_action_class
 
-_logger = get_kpm_logger()
-
 
 class ScAgentAbstract(ABC):
     def __init__(self, event_element: ScAddr, event_type: ScEventType) -> None:
         self._event_element = event_element
         self._event_type = event_type
         self._event: Optional[ScEvent] = None
+        self._logger = getLogger(f"{self.__module__}.{self.__class__.__name__}")
```

Comment:
should _self.logger_ name be private? I think it should be public and used whenever user want it to use

Reply:
Yeah, great idea

Reply:
Only ScAgent of also ScModule and ScServer?

Reply:
I think it should be everywhere, we didn't know all wishes of the user 🐱